### PR TITLE
Feature/delete permenently top speaker

### PR DESCRIPTION
**Pull request by dilafar**

### CODE REVIEW

**Review by KaveenLaksitha — CHANGES_REQUESTED** [review]
Please check the spelling of the word "Permanent".

**Review by rusiruavb — APPROVED** [review]
LGTM

File: src/store/top-speaker-store/topSpeakerActions.ts [hunk 1]
```diff
@@ -43,6 +43,12 @@ export const setTopSpeakerId = (topSpeakerId: string) => {
 	};
 };
 
+export const permanentDeleteTopSpeaker = (topSpeakerId: string) => {
+	return {
+		type: topSpeakerActionTypes.PERMANENT_DELETE_TOP_SPEAKER,
```
Comment (senuravihanjayadeva) [inline src/store/top-speaker-store/topSpeakerActions.ts:48]: Can we use the DELETE_TOP_SPEAKER action type. Refer Event Actions

Comment (senuravihanjayadeva) [reply src/store/top-speaker-store/topSpeakerActions.ts:48]: And do the rest of the modifications

File: src/store/top-speaker-store/topSpeakerReducer.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ const initialState: ITopSpeakerStore = {
 	addTopSpeaker: null,
 	updatedTopSpeaker: null,
 	deletedTopSpeaker: null,
+	permanentDeletedTopSpeaker: null,
```
Comment (senuravihanjayadeva) [inline src/store/top-speaker-store/topSpeakerReducer.ts:12]: No need you can use deletedTopSpeaker

File: src/store/top-speaker-store/topSpeakerReducer.ts [hunk 3]
```diff
@@ -19,6 +20,7 @@ const topSpeakerReducer = (state = initialState, action: any) => {
 		case `${TopSpeakerActionTypes.GET_TOP_SPEAKER}_PENDING`:
 		case `${TopSpeakerActionTypes.GET_ALL_TOP_SPEAKERS}_PENDING`:
 		case `${TopSpeakerActionTypes.DELETE_TOP_SPEAKER}_PENDING`:
+		case `${TopSpeakerActionTypes.PERMANENT_DELETE_TOP_SPEAKER}_PENDING`:
```
Comment (senuravihanjayadeva) [inline src/store/top-speaker-store/topSpeakerReducer.ts:23]: No need

File: src/store/top-speaker-store/topSpeakerReducer.ts [hunk 4]
```diff
@@ -43,13 +45,17 @@ const topSpeakerReducer = (state = initialState, action: any) => {
 		case `${TopSpeakerActionTypes.DELETE_TOP_SPEAKER}_FULFILLED`:
 			let deletedTopSpeaker = action.payload.data;
 			return { ...state, loading: false, deletedTopSpeaker };
+		case `${TopSpeakerActionTypes.PERMANENT_DELETE_TOP_SPEAKER}_FULFILLED`:
```
Comment (senuravihanjayadeva) [inline src/store/top-speaker-store/topSpeakerReducer.ts:48]: No need

File: src/store/top-speaker-store/topSpeakerReducer.ts [hunk 5]
```diff
@@ -43,13 +45,17 @@ const topSpeakerReducer = (state = initialState, action: any) => {
...
+			let permanentDeletedTopSpeaker = action.payload.data;
+			return { ...state, loading: false, permanentDeletedTopSpeaker };
 
 		case `${TopSpeakerActionTypes.CREATE_TOP_SPEAKER}_REJECTED`:
 		case `${TopSpeakerActionTypes.GET_TOP_SPEAKER}_REJECTED`:
 		case `${TopSpeakerActionTypes.GET_ALL_TOP_SPEAKERS}_REJECTED`:
 		case `${TopSpeakerActionTypes.GET_DELETED_TOP_SPEAKERS}_REJECTED`:
 		case `${TopSpeakerActionTypes.UPDATE_TOP_SPEAKER}_REJECTED`:
 		case `${TopSpeakerActionTypes.DELETE_TOP_SPEAKER}_REJECTED`:
+		case `${TopSpeakerActionTypes.PERMANENT_DELETE_TOP_SPEAKER}_REJECTED`:
```
Comment (senuravihanjayadeva) [inline src/store/top-speaker-store/topSpeakerReducer.ts:58]: No need

File: src/pages/top-speaker/permenent-delete/index.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+	useEffect(() => {
+		dispatch(getDeletedTopSpeakers());
+		dispatch(setTopSpeakerId(""));
+
+		if (state.permanentDeletedTopSpeaker) {
+			toastNotification("Top Speaker removed successfully", "success");
+		}
+
+		closeModal();
+	}, [state.permanentDeletedTopSpeaker, dispatch]);
```
Comment (senuravihanjayadeva) [inline src/pages/top-speaker/permenent-delete/index.tsx:35]: state.permanentDeletedTopSpeaker -> state.deletedTopSpeaker

File: src/interfaces/ITopSpeaker.ts [hunk 7]
```diff
@@ -23,6 +23,7 @@ interface ITopSpeakerStore {
 	addTopSpeaker: ITopSpeaker | null;
 	updatedTopSpeaker: ITopSpeaker | null;
 	deletedTopSpeaker: ITopSpeaker | null;
+	permanentDeletedTopSpeaker: ITopSpeaker | null;
```
Comment (senuravihanjayadeva) [inline src/interfaces/ITopSpeaker.ts:26]: No need

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

**Pull request merged**